xapian_backend.py

```diff
@@ -102,31 +101,36 @@ def __call__(self, begin, end):
             if field_dict['field_name'] == field_name:
                 field_type = field_dict['type']
 
-                if not begin:
+                if begin:
+                    if field_type == 'float':
+                        begin = _term_to_xapian_value(float(begin), field_type)
+                    elif field_type == 'integer':
+                        begin = _term_to_xapian_value(int(begin), field_type)
+                else:
                     if field_type == 'text':
                         begin = 'a'  # TODO: A better way of getting a min text value?
-                    elif field_type == 'integer':
-                        begin = -sys.maxsize - 1
-                    elif field_type == 'float':
-                        begin = float('-inf')
+                    elif field_type in ('float', 'integer'):
+                        # floats and ints are both serialised using xapian.sortable_serialise
+                        # so we can use -Infinity as the lower bound for both of them.
+                        begin = _term_to_xapian_value(float('-inf'), field_type)
                     elif field_type == 'date' or field_type == 'datetime':
                         begin = '00010101000000'
-                elif end == '*':
+
+                if end == '*':
                     if field_type == 'text':
                         end = 'z' * 100  # TODO: A better way of getting a max text value?
-                    elif field_type == 'integer':
-                        end = sys.maxsize
-                    elif field_type == 'float':
-                        end = float('inf')
+                    elif field_type in ('float', 'integer'):
+                        # floats and ints are both serialised using xapian.sortable_serialise
+                        # so we can use +Infinity as the upper bound for both of them.
+                        end = _term_to_xapian_value(float('inf'), field_type)
                     elif field_type == 'date' or field_type == 'datetime':
                         end = '99990101000000'
+                else:
+                    if field_type == 'float':
+                        end = _term_to_xapian_value(float(end), field_type)
+                    elif field_type == 'integer':
+                        end = _term_to_xapian_value(int(end), field_type)
```

Comment:
I think we can also simplify the logic for the other field types here. [`ValueRangeProcessors`](https://xapian.org/docs/valueranges.html) (which should maybe just be `RangeProcessors` now?) can return empty values to represent open ranges. In that case we wouldn't have to make up arbitrary lower and upper bounds for text and date values.